components/gfx/text/util.rs

```diff
@@ -38,7 +38,7 @@ pub fn transform_text(text: &str,
                     output_text.push(ch);
                 }
             }
-            text.len() > 0 && is_in_whitespace(text.char_at_reverse(0), mode)
+            false
```

Comment:
This fixes https://github.com/servo/servo/blob/master/tests/wpt/css-tests/css21_dev/html4/white-space-mixed-002.htm and some related issues.  The return value here basically means "should leading whitespace be discarded from the next fragment", and the answer is always no in `white-space: pre` or `white-space: pre-wrap` mode.